src/api/auth/src/user.js

```diff
@@ -1,23 +1,98 @@
-// Regular (non-admin) User
+// A User represents a Seneca SSO Authenticated user who might also be a Telescope user.
 class User {
-  constructor(name, email, id, nameID, nameIDFormat) {
-    this.name = name;
-    this.email = email;
-    this.id = id;
-    this.nameID = nameID;
-    this.nameIDFormat = nameIDFormat;
-    this.isAdmin = false;
+  constructor(senecaProfile, telescopeProfile) {
+    // TODO: what to do about id?  Hash it like before or use email? hash(profile.nameID)
+    // this.id = id;
+
+    // All authenticated users are Seneca Users
+    this.seneca = {
+      'http://schemas.microsoft.com/identity/claims/displayname':
+        senecaProfile['http://schemas.microsoft.com/identity/claims/displayname'],
+      'http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress':
+        senecaProfile['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress'],
```

Comment:
Are these links supposed to work when we access them? If not, how do they specify the data type?

Reply:
They aren't links, they are URIs for specific schema property names.  It's a way for there to be less ambiguity about what "email" or "displayname" means: the URI version is very specific.  NOTE: I'm using what we get back from Microsoft, not choosing these names.